src/Rector/ClassMethod/ArgumentValueResolverToValueResolverRector.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Symfony\Rector\ClassMethod;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\BinaryOp\Identical;
+use PhpParser\Node\Expr\BinaryOp\NotIdentical;
+use PhpParser\Node\Expr\ConstFetch;
+use PhpParser\Node\Name;
+use PhpParser\Node\Name\FullyQualified;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\If_;
+use PhpParser\Node\Stmt\Return_;
+use Rector\Core\Rector\AbstractRector;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\Controller\ValueResolverInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Symfony\Tests\Rector\ClassMethod\ArgumentValueResolverToValueResolverRector\ArgumentValueResolverToValueResolverRectorTest
+ */
+final class ArgumentValueResolverToValueResolverRector extends AbstractRector
+{
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        // Check if the class implements ArgumentValueResolverInterface
+        $isArgumentValueResolver = false;
+        foreach ($node->implements as $key => $interface) {
+            if ($interface->toString() === ArgumentValueResolverInterface::class) {
+                $node->implements[$key] = new FullyQualified(ValueResolverInterface::class);
+                $isArgumentValueResolver = true;
+                break;
+            }
+        }
+
+        // If it doesn't implement ArgumentValueResolverInterface, skip
+        if (! $isArgumentValueResolver) {
+            return null;
+        }
+
+        $isIdentical = true;
+        foreach ($node->getMethods() as $key => $classMethod) {
```

Comment:
This makes it work but right now its a bit messy and PHPstan also doesn't like i should have a look how to improve this part.

Reply:
@stefantalen maybe you have some suggestions for me? :)

Reply:
Not really, maybe moving some logic to _private_ functions would help.

Reply:
Yeah thanks for the tip i did that :)